test/module/irohad/ametsuchi/postgres_block_storage_test.cpp

```diff
@@ -34,43 +41,79 @@ class PostgresBlockStorageTest : public AmetsuchiTest {
 
  protected:
   void SetUp() override {
-    AmetsuchiTest::SetUp();
+    PostgresOptions options(
+        pgopt_, PgConnectionInit::kDefaultDatabaseName, storage_logger_);
+
+    PgConnectionInit connection_init;
+    connection_init
+        .createDatabaseIfNotExist(options.dbname(),
+                                  options.optionsStringWithoutDbName())
+        .match([](auto &&val) {},
+               [&](auto &&error) {
+                 storage_logger_->error("Database creation error: {}",
+                                        error.error);
+                 std::terminate();
+               });
+
+    auto reconnection_strategy_factory =
+        std::make_unique<iroha::ametsuchi::KTimesReconnectionStrategyFactory>(
+            0);
+
+    auto pool = connection_init.prepareConnectionPool(
+        *reconnection_strategy_factory,
+        options,
+        pool_size_,
+        getTestLoggerManager()->getChild("Storage"));
+
+    if (auto e = boost::get<iroha::expected::Error<std::string>>(&pool)) {
+      storage_logger_->error("Pool initialization error: {}", e->error);
+      std::terminate();
+    }
+
+    pool_wrapper_ = std::move(
+        boost::get<iroha::expected::Value<std::shared_ptr<PoolWrapper>>>(pool)
+            .value);
 
     auto validator = std::make_unique<MockBlockIValidator>();
     auto proto_validator = std::make_unique<MockBlockPValidator>();
 
-    block_factory_ =
-        std::make_shared<shared_model::proto::ProtoTransportFactory<
-            shared_model::interface::Block,
-            shared_model::proto::Block>>(std::move(validator),
-                                         std::move(proto_validator));
+    block_factory_ = std::make_shared<shared_model::proto::ProtoBlockFactory>(
+        std::move(validator), std::move(proto_validator));
 
-    sql_ = std::make_unique<soci::session>(*soci::factory_postgresql(), pgopt_);
     block_storage_ =
-        PostgresBlockStorageFactory(
-            *sql_, block_factory_, getTestLogger("PostgresBlockStorage"))
+        PostgresBlockStorageFactory(pool_wrapper_,
+                                    block_factory_,
+                                    [&]() { return test_table_; },
+                                    getTestLogger("PostgresBlockStorage"))
             .create();
-    *sql_ << "CREATE TABLE IF NOT EXISTS blocks (height bigint PRIMARY KEY, "
-             "block_data text not null);";
+    soci::session sql(*pool_wrapper_->connection_pool_);
+    sql << "CREATE TABLE IF NOT EXISTS " << test_table_
+        << " (height bigint PRIMARY KEY, "
+           "block_data text not null);";
   }
 
   void TearDown() override {
-    *sql_ << "DROP TABLE IF EXISTS blocks;";
-    sql_->close();
-    AmetsuchiTest::TearDown();
+    soci::session sql(*pool_wrapper_->connection_pool_);
+    sql << "DROP TABLE IF EXISTS " << test_table_;
```

Comment:
feels like `PostgresTemporaryBlockStorage`, could we use it?

Reply:
Same here.